cmake/partition_manager.cmake

```diff
@@ -4,6 +4,60 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+# set_dfu_hex_offset([EXT_START <address> EXT_BASE <address>])
+function(set_dfu_hex_offset)
```

Comment:
I feel this function is very messy.
It's not really a function that you can reuse anywhere, because it relies intensively on variables in outer scope (all th `PM_` vars).

Also it has `get_shared()` calls.

I think we should consider to clean this up, and in that process consider if populating the whole CMake workspace with `PM_` vars is the best way.

If those were properties on a target, then we could simply pass that target to this function, and it would become more generic.

I acknowledge this is outside the scope of this PR, but could you create some followup task.
We might already have a task for something like this in relation to some other partition manager work, if I recall correctly.

Reply:
Good point, this might fit in well with the plans to execute partition manager early configure time so that we have the variables available earlier in the process. Due to other refactoring it might be better to not have this as a function now, I will make a proposal with this in-lined as it was before.